### PR TITLE
ci: add workflow automation to ffmpeg-sdk

### PR DESCRIPTION
**Pull request by soumyaDghosh**

### CODE REVIEW

File: .github/workflows/release-content-to-candidate.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    needs: get-architectures
+    runs-on: ubuntu-latest
+    environment: "2204 Branch"
+    strategy:
+      matrix:
+        architecture: ${{ fromJSON(needs.get-architectures.outputs.architectures-list) }}
+    steps:
+      - name: 🚢 Release to latest/candidate
+        id: build-amd64
+        uses: soumyaDghosh/ci/release-to-candidate@main
```
Comment (jnsgruk) [inline .github/workflows/release-content-to-candidate.yml]:
```suggestion
        uses: snapcrafters/ci/release-to-candidate@main
```

File: .github/workflows/release-sdk-to-candidate.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    name: 🚢 Release to latest/candidate
+    needs: get-architectures
+    runs-on: ubuntu-latest
+    environment: "2204 Branch"
+    strategy:
+      matrix:
+        architecture: ${{ fromJSON(needs.get-architectures.outputs.architectures-list) }}
+    steps:
+      - name: 🚢 Release to latest/candidate
+        uses: soumyaDghosh/ci/release-to-candidate@main
```
Comment (jnsgruk) [inline .github/workflows/release-sdk-to-candidate.yml]:
```suggestion
        uses: snapcrafters/ci/release-to-candidate@main
```

File: .github/workflows/promote-to-stable.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      && contains(github.event.comment.body, '/promote ')
+      && contains(github.event.issue.body, '`ffmpeg-2204-sdk`')
+      && contains(github.event.*.labels.*.name, 'testing')
+    steps:
+      - name: ⬆️ Promote to stable
+        uses: snapcrafters/ci/promote-to-stable@main
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          store-token: ${{ secrets.SNAP_STORE_STABLE }}
+          snapcraft-project-root: "ffmpeg-2204-sdk"
```
Comment (jnsgruk) [inline .github/workflows/promote-to-stable.yml]: Nit: this file need a newline at the end :)

File: .github/workflows/release-content-to-candidate.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+  issues: write
+
+jobs:
+  get-architectures:
+    if: ${{ github.event.workflow_run.conclusion == 'success' || github.event_name == 'workflow_dispatch' }}
```
Comment (jnsgruk) [inline .github/workflows/release-content-to-candidate.yml:21]: What is this doing at this point in the workflow? It looks redundant to me? :)

**Review by jnsgruk — CHANGES_REQUESTED** [review]
A couple of small changes -- could you also squash this down into a couple of well labelled commits? Let me know if you need a hand.

File: .github/workflows/release-content-to-candidate.yml (hunk 4 above)
Comment (soumyaDghosh) [reply .github/workflows/release-content-to-candidate.yml:21]: This workflow will run when the `release-sdk-to-candidate` is successful. Now if the release fails, this condition is used to check that.

File: .github/workflows/release-sdk-to-candidate.yml (hunk 2 above)
Comment (soumyaDghosh) [reply .github/workflows/release-sdk-to-candidate.yml]: Waiting for the CI PR to merge before changing this.

File: .github/workflows/release-content-to-candidate.yml (hunk 1 above)
Comment (soumyaDghosh) [reply .github/workflows/release-content-to-candidate.yml]: same, waiting for the CI PR to merge.

File: .github/workflows/promote-to-stable.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+jobs:
+  promote-content:
+    name: ⬆️ Promote Content to stable
+    environment: "2204 Branch"
+    runs-on: ubuntu-latest
+    if: |
+      ( !github.event.issue.pull_request )
+      && contains(github.event.comment.body, '/promote ')
+      && contains(github.event.issue.body, '`ffmpeg-2204`')
```
Comment (jnsgruk) [inline .github/workflows/promote-to-stable.yml]:
```suggestion
      && contains(github.event.issue.body, '`ffmpeg-2204 `')
```

If you add a space here, you'll prevent this from triggering when the command is issued for the SDK snap :)

**Review by jnsgruk — DISMISSED** [review]
Looks good, this has been a long one so thanks for sticking with it! I'll merge now with one very minor change and we can monitor how it goes.

**Review by jnsgruk — APPROVED** [review]

**Comment by jnsgruk** [comment]
Merged in https://github.com/snapcrafters/ffmpeg-sdk/commit/c457df3f633769e8c8f87dcc57e6518e1252a942

**Pull request closed without merging**